repos/system_upgrade/common/actors/checkconsumedassets/libraries/check_consumed_assets.py

```diff
@@ -0,0 +1,162 @@
+import re
+from collections import defaultdict, namedtuple
+
+from leapp import reporting
+from leapp.libraries.common.config import get_consumed_data_stream_id
+from leapp.libraries.common.fetch import ASSET_PROVIDED_DATA_STREAMS_FIELD
+from leapp.libraries.stdlib import api
+from leapp.models import ConsumedDataAsset
+
+
+def compose_summary_for_incompatible_assets(assets, incompatibility_reason):
+    if not assets:
+        return []
+
+    summary_lines = ['The following assets are {reason}'.format(reason=incompatibility_reason)]
+    for asset in assets:
+        if asset.provided_data_streams is None:  # Assets with missing streams are placed only in .outdated bucket
+            details = (' - The asset {what_asset} is missing information about provided data streams '
+                       'in its metadata header')
+            details = details.format(what_asset=asset.filename)
+        else:
+            article, multiple_suffix = ('the ', '') if len(asset.provided_data_streams) == 1 else ('', 's')
+            details = ' - The asset {what_asset} provides data {article}stream{mult_suffix} {provided_streams}'
```

Comment:
Doesn't provide a nice message in all cases, for example for a single from-the-future asset 42.42:
```
"summary": "The currently installed Leapp consumes data stream 1.0, but the following assets provide different streams:\nThe following assets are intended for a newer leapp\n - The asset pes-events.json provides data the stream 42.42"
```
Note the "The asset pes-events.json provides **data the stream 42.42**"